data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -162,11 +162,13 @@ private static DataServiceException createExceptionFromResponse(String url, Http
                     return new DataServiceException(url, message, formattedException,
                             formattedException.getApiError().isPermanent());
                 } catch (Exception ignored) {
+                    // Exception doesn't follow the OneAPI error format
+                    return new DataServiceException(url, errorFromResponse, false);
```

Comment:
1. Usually it's not a healthy pattern to communicate between code parts using exceptions. i.e. we assume this is an API error then we try and fail - can we avoid this?
2. Can we try get the actual message instead of printing {"message": "..."}

Reply:
1. I changed the code. Now I parse the json and see from there, but we do still have this problem because you can't know if the json is valid until you parse it. I added a comment to explain.
2. Done